modguard/show.py

```diff
@@ -0,0 +1,42 @@
+import yaml
+from modguard.core.boundary import BoundaryTrie
+from typing import Any, Dict
+
+# This type hint only works on more recent versions
+# result_dict: TypeAlias = dict[str, str | bool | 'result_dict']
+
+def boundary_trie_to_dict(boundary_trie: BoundaryTrie) -> Dict[str, Any]:
+    result: Dict[str, Any] = dict()
+    for node in boundary_trie:
+        path = node.full_path
+        if path == "":
+            path = "root"
+        sections = path.split(".")
+        pointer: Dict[str, Any] = result
+        for section in sections:
+            if section not in pointer:
+                pointer[section] = dict()
+            pointer = pointer[section]
+        pointer["is_boundary"] = True
+
+        for member in node.public_members.keys():
+            pointer: Dict[str, Any] = result
+            sections = member.split(".")
```

Comment:
Since the root boundary `__init__.py` may be marked `public()` and would then have a public member with an empty path, we probably want to explicitly handle that case and assign `member` to "root" like you do in the previous loop.

Reply:
We could also ignore the root boundary altogether, not sure it adds anything to the output